addons/point_of_sale/upgrades/1.0.2/post-deduplicate-uuids.py

```diff
@@ -0,0 +1,35 @@
+def migrate(cr, version):
+    """
+    When UUIDs were introduced for POS [order, order.line, payment] records,
+    they were initially generated on the backend, defined as the default value
+    on the column. The issue with this is that during the upgrade to 18.0+
+    from a version < 18.0, this default value is determined once for the column
+    and then applied to all records, resulting in one UUID, duplicated across 
+    all existing POS records.
+    
+    This migration fixes the issue by generating a new UUID for each record
+    that has the same UUID as another record. Specifically, it does this for
+    the following tables:
+    - pos_order
+    - pos_order_line
+    - pos_payment
+    """
+    def deduplicate_uuids(table):
+        # Use gen_random_uuid() to generate a new v4 UUID on the fly
+        query = f"""
+            WITH duplicate_uuids AS (
+                SELECT uuid
+                    FROM {table}
+                    GROUP BY uuid
+                HAVING COUNT(*) > 1
+                )
+            UPDATE {table}
+                SET uuid = gen_random_uuid()
```

Comment:
This is not supported in PG12, which is the minimal PG version for Odoo 18
https://www.postgresql.org/docs/13/functions-uuid.html